code/game/objects/explosion.dm

```diff
@@ -44,6 +44,14 @@ proc/explosion(turf/epicenter, devastation_range, heavy_impact_range, light_impa
 					far_volume += (dist <= far_dist * 0.5 ? 50 : 0) // add 50 volume if the mob is pretty close to the explosion
 					M.playsound_local(epicenter, 'sound/effects/explosionfar.ogg', far_volume, 1, frequency, falloff = 5)
 
+		var/close = range(world.view+round(devastation_range,1), epicenter)
+		// to all distanced mobs play a different sound
+		for(var/mob/M in world) if(M.z == epicenter.z) if(!(M in close))
+			// check if the mob can hear
+			if(M.ear_deaf <= 0 || !M.ear_deaf)

```

Comment:
While the last time was complete nonsense code that didn't deserve to compile, you CAN do this in a single line+if.
```if((M.ear_deaf <= 0 || !M.ear_deaf)) && !istype(M.loc, /turf/space))``
Though, you most likely want to swap your current ```M.loc``` for ```get_turf(M)```

As to WHY the last try compiled, I'd assume (token wise) it became:
```
if(M.ear_deaf <= 0 || !M.ear_deaf)
    if(!istype(M.loc, /turf/space))
        ...
```
Which, amusingly, is what you have written here.